### PR TITLE
Series: display number of documents

### LINKED ISSUE #474: Series: show number of documents
In the Series search results and details page, show the number of documents that the series has.

Both on the frontsite and in the backoffice
Consult the task before starting!

Request changes:
semanic-ui <Label> to highlight the number of volumes
revert the order of words: eg. 1 volume, 2 volumes
`online` `<Label>` when we have the `access_urls` field (not empty) on the series

### PR DESCRIPTION
**Pull request by KonstantinaStoikou**
On the frontsite display the number of documents in the details page of every series.
details_frontsite

 On the backoffice display the same in the details page and in the search page.
details_backoffice
search_backoffice

After clicking in the link in the search page of backoffice:
after_click_backoffice

Closes #474

### CODE REVIEW

File: ui/src/pages/backoffice/Series/SeriesDetails/SeriesDetails.js [hunk 1]
```diff
@@ -41,10 +41,18 @@ export default class SeriesDetails extends Component {
 
   seriesPanels = () => {
     const { data } = this.props;
+    let literatureNum = 0
```
Comment (ntarocco) [inline ui/src/pages/backoffice/Series/SeriesDetails/SeriesDetails.js]: If I am not wrong, you can have only one or the other, so what about checking both and summing? One should be always 0...

You can also use (to be tested!!!) [lodash get]() here:
```
const serialsCount = _get(data, 'metadata.relations_metadata.serial', []).length
const monographsCount = _get(data, 'metadata.relations_metadata.multipart_monograph', []).length
const documentsCount = serialsCount + monographsCount
```

and then below:
```
title: 'Literatures in this series (' + documentsCount + ')',
```

WDYT?

File: ui/src/pages/backoffice/Series/SeriesSearch/SeriesListEntry.js [hunk 2]
```diff
@@ -33,6 +33,17 @@ export class SeriesListEntry extends Component {
 
   renderRelations = () => {
```
Comment (ntarocco) [inline ui/src/pages/backoffice/Series/SeriesSearch/SeriesListEntry.js:35]: @kprzerwa 
I think we should remove all links to search query here. For example, I went to one that had 5 editions and the link is: `.../documents?q=pid: 6vptg-ty662 OR pid: 9msba-sxt46 OR pid: r43kz-hdv89 OR pid: t24n3-ht166 OR pid: hnsq8-krm96`.
The list of `OR` can be potentially massive.

IMHO, it is important to show the text, e.g `5 editions`, then when you go inside the details page, you can see the full list in the relations boxes...

So in this case, I would just count as suggested just above and show `10 literatures` text and remove the link.

WDYT?

File: ui/src/pages/frontsite/components/Series/SeriesLiteratureSearch/SeriesLiteratureSearch.js [hunk 3]
```diff
@@ -49,6 +49,14 @@ export class SeriesLiteratureSearch extends React.Component {
 
   render() {
     const { metadata } = this.props;
+    let literatureNum
+    if (metadata.relations_metadata) {
```
Comment (ntarocco) [inline ui/src/pages/frontsite/components/Series/SeriesLiteratureSearch/SeriesLiteratureSearch.js]: Same suggestion above if it makes any sense to you.

**Review by ntarocco — COMMENTED** [review]
@KonstantinaStoikou could you please add screenshots of the modifications? Normally when we touch UI we try to add screenshots in the comment of the PR so it is easier to understand and see the change! Thanks! :)

File: ui/src/pages/backoffice/Series/SeriesDetails/SeriesDetails.js (hunk 1 above)
Comment (topless) [inline ui/src/pages/backoffice/Series/SeriesDetails/SeriesDetails.js]:
```suggestion
    let literatureNum = 0;
```

File: ui/src/pages/backoffice/Series/SeriesDetails/SeriesDetails.js [hunk 4]
```diff
@@ -41,10 +41,18 @@ export default class SeriesDetails extends Component {
...
+    if (data.metadata && data.metadata.relations_metadata) {
```
Comment (topless) [inline ui/src/pages/backoffice/Series/SeriesDetails/SeriesDetails.js]:
```js
literatureNum = 0
if (data.metadata && data.metadata.relations_metadata) {
  literatureNum = data.metadata.relations_metadata.serial ?
    data.metadata.relations_metadata.serial.length :
    data.metadata.relations_metadata.multipart_monograph.length;
}
```

File: ui/src/pages/backoffice/Series/SeriesDetails/SeriesDetails.js [hunk 5]
```diff
@@ -41,10 +41,18 @@ export default class SeriesDetails extends Component {
...
+        data.metadata.relations_metadata.serial.length :
+        data.metadata.relations_metadata.multipart_monograph.length
+    } else {
+      literatureNum = 0
+    }
     const panes = [
       {
         key: 'series-documents',
-        title: 'Literature in this series',
+        title: 'Literature in this series (' + literatureNum + ')',
```
Comment (topless) [inline ui/src/pages/backoffice/Series/SeriesDetails/SeriesDetails.js]: This is the new madness in string interpolation :)
```suggestion
        title: `Literature in this series (${literatureNum})`,
```

File: ui/src/pages/backoffice/Series/SeriesSearch/SeriesListEntry.js [hunk 6]
```diff
@@ -33,6 +33,17 @@ export class SeriesListEntry extends Component {
...
     const { series } = this.props;
+    let literatureNum;
```
Comment (topless) [inline ui/src/pages/backoffice/Series/SeriesSearch/SeriesListEntry.js]: You can initialize it here `literatureNum = 0` and skip the `else` part in your condition.

File: ui/src/pages/backoffice/Series/SeriesSearch/SeriesListEntry.js [hunk 7]
```diff
@@ -83,6 +94,22 @@ export class SeriesListEntry extends Component {
                 )}
               </List.Item>
             )}
+
+            {!_isEmpty(series.metadata.relations_metadata) && (
+              <List.Item>
+                <List.Content>
+                  <Link
+                    to={BackOfficeRoutes.documentsListWithQuery(
+                      'relations.' + searchStr + '.pid:' + series.metadata.pid
```
Comment (topless) [inline ui/src/pages/backoffice/Series/SeriesSearch/SeriesListEntry.js]: Lets use the string madness from above

File: ui/src/pages/backoffice/Series/SeriesSearch/SeriesListEntry.js [hunk 8]
```diff
@@ -83,6 +94,22 @@ export class SeriesListEntry extends Component {
...
+                    )}
+                  >
+                    See all volumes in this series ({literatureNum}){' '}
```
Comment (topless) [inline ui/src/pages/backoffice/Series/SeriesSearch/SeriesListEntry.js]:
```suggestion
                    See all volumes in this series ({literatureNum})
```

File: ui/src/pages/frontsite/components/Series/SeriesLiteratureSearch/SeriesLiteratureSearch.js [hunk 9]
```diff
@@ -49,6 +49,14 @@ export class SeriesLiteratureSearch extends React.Component {
 
   render() {
     const { metadata } = this.props;
+    let literatureNum
```
Comment (topless) [inline ui/src/pages/frontsite/components/Series/SeriesLiteratureSearch/SeriesLiteratureSearch.js]: Same as above ;)

**Review by topless — COMMENTED** [review]
Great stuff! Small thingies here and there! It seems that `Prettier` is either not configured properly or not enabled.

**Comment by ntarocco** [comment]
I was about to merge it, but you add the number of documents also in the frontsite search page? @kprzerwa WDYT?

Please add in the commit message `closes #474` if not done (I didn't see it)

**Pull request merged**